src/modules/pos/validator/components/ValidatorsMonitorView/Validators.js

```diff
@@ -32,6 +33,27 @@ import styles from './Validators.css';
 const ValidatorActionButton = ({ address, isValidator }) => {
   const { t } = useTranslation();
   const tokenBalances = useTokenBalances({ options: { enabled: !isValidator || !address } });
+  const hasTokenBalances = tokenBalances.data?.data?.some(
+    ({ availableBalance }) => !!BigInt(availableBalance)
+  );
+
+  const { hasSufficientBalanceForFee, feeToken } = useValidateFeeBalance();
+
+  const getInSuffienctBalanceMessage = () => {
+    if (!hasTokenBalances) {
+      return {
+        message: t('Token balance is not enough to stake a validator.'),
```

Comment:
Cant we abstract these messages somehow. A lot of duplication  of `if (!hasSufficientBalanceForFee) {`,  `if (!hasTokenBalances) {` varies slightly between components. But feels like it could be abstracted to a hook `useInSufficientBalanceMessage(noTokenBalanceSuffix)` or we can use a const `IN_SUFFICIENT_BALANCE_MESSAGES = {...}`

`const data = useInSufficientBalanceMessage('to stake a validator.')`
Or we can pass in domain 🤔 
`const data = useInSufficientBalanceMessage(VALIDATOR)`